### PR TITLE
Use Appium to get iOS E2E logs

### PR DESCRIPTION
**Pull request by nickdowell**
## Goal

Make it easier to investigate test failures by including device syslog in `maze_output`.

## Changeset

Uses Appium's [get_log](https://appium.io/docs/en/commands/session/logs/get-log/) method to fetch the device log for each scenario.

@twometresteve once proven, this probably should find its way into maze-runner.
On Android, I think `get_log('logcat')` should work.

## Testing

Verified logs are populated when running locally.

### CODE REVIEW

**Review by twometresteve — COMMENTED** [review]
Were you expecting at least one of the pipeline steps to upload files as buildkite artifacts?

**Comment by nickdowell** [comment]
> Were you expecting at least one of the pipeline steps to upload files as buildkite artifacts?

Only failed scenarios have their logs uploaded, so I wasn't expecting to see any in a successful build.

**Review by twometresteve — APPROVED** [review]
Nice

File: features/support/env.rb [hunk 1]
```diff
@@ -130,17 +119,17 @@ def skip_below(os, version)
       FileUtils.mv '/tmp/kscrash.log', path
     end
   when 'ios'
-    if $logger_pid
-      Process.kill 'TERM', $logger_pid
-      Process.waitpid $logger_pid
-      $logger_pid = nil
-      FileUtils.mv 'device.log', path
+    # get_log can be slow (1 or 2 seconds) on device farms
+    if scenario.failed || Maze.config.farm == :local
```
Comment (twometresteve) [inline features/support/env.rb:123]:
```suggestion
    if scenario.failed? || Maze.config.farm == :local
```

**Review by twometresteve — APPROVED** [review]

**Pull request merged**